subiquity/server/controllers/shutdown.py

```diff
@@ -127,7 +128,10 @@ async def copy_logs_to_target(self, context):
             log.exception("saving journal failed")
 
     @with_context(description='mode={self.mode.name}')
-    def shutdown(self, context):
+    async def shutdown(self, context):
+        await self.app.hub.abroadcast(InstallerChannels.PRE_SHUTDOWN)
```

Comment:
[how do we guarantee / is there a guarantee] that FilesystemController._pre_shutdown can run before we exit the application?

Reply:
Create an ephemeral systemd unit that depends on shutdown.target or whatever it is?